Fix CRLF line-ending parsing for comments.

Eljay opened the pull request:
Fixes #25182, parser didn't account for \r\n in regular comments, only doc-comments.

rust-highfive commented:
Thanks for the pull request, and welcome! The Rust team is excited to review your changes, and you should hear from @huonw (or someone else) soon.

If any changes to this PR are deemed necessary, please add them as extra commits. This ensures that the reviewer can see what has changed since they last reviewed the code. The way Github handles out-of-date commits, this should also make it reasonably obvious what issues have or haven't been addressed. Large or tricky changes may require several passes of review and changes.

Please see [CONTRIBUTING.md](https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md) for more information.

alexcrichton commented on src/libsyntax/parse/lexer/mod.rs:
```diff
@@ -403,45 +403,47 @@ impl<'a> StringReader<'a> {
-                        return Some(TokenAndSpan {
+                        Some(TokenAndSpan {
                             tok: token::Comment,
                             sp: codemap::mk_sp(start_bpos, self.last_pos)
```
I think this `start_bpos` is off by one perhaps? It was originally `last-BytesPos(2)` but now I think it's `last-BytePos(3)`

alexcrichton commented:
Thanks! Would it be possible to add a test for this as well?

Eljay commented:
Ok the bytepos should now be consistent with how it was previously, and I've fixed up the error message change that broke some tests. I've also added a test to check that the newlines are stripped but not sure if this is the best way to do it.

alexcrichton commented:
Hm I believe though that in the first case (a doc comment) the byte pos was subtracted by 3, and in this case it's now subtracted by only two? Perhaps some more assertions could be added to the test that was added about the spans being returned as well?

Eljay commented:
Previously the byte pos was `pos - 3` for doc comments and `last_pos - 2` otherwise to account for characters wider than 1 byte in comments (e.g. starting with `//<wide char>`). So `last_pos - 2` should be correct independent of the byte width of the current character. Correct?

alexcrichton commented:
The branch for doc comments has `last_pos - 3` previously, and the branch for regular comments had `last_pos - 2`, but now both branches use `last_pos - 2` which is all I'm worried about

Eljay commented:
It previously used `pos - 3`, not `last_pos - 3`. Seeing as the third character for doc comments is always `/` or `!`, which both have a byte width of 1, `last_pos - 2` should always be equal to `pos - 3` unless I'm going mad. I will make it explicit anyway to be sure.

alexcrichton commented:
Oh wow I totally missed the `pos` vs `last_pos` distinction, this is subtle! Anyway I think current current version looks good to me, so thanks!

@bors: r+ 2dcc200

bors commented:
:hourglass: Testing commit 2dcc200 with merge b948d81...

bors commented:
:sunny: Test successful - [auto-linux-32-nopt-t](http://buildbot.rust-lang.org/builders/auto-linux-32-nopt-t/builds/4965), [auto-linux-32-opt](http://buildbot.rust-lang.org/builders/auto-linux-32-opt/builds/4972), [auto-linux-64-nopt-t](http://buildbot.rust-lang.org/builders/auto-linux-64-nopt-t/builds/4928), [auto-linux-64-opt](http://buildbot.rust-lang.org/builders/auto-linux-64-opt/builds/4952), [auto-linux-64-x-android-t](http://buildbot.rust-lang.org/builders/auto-linux-64-x-android-t/builds/4922), [auto-mac-32-opt](http://buildbot.rust-lang.org/builders/auto-mac-32-opt/builds/4997), [auto-mac-64-nopt-t](http://buildbot.rust-lang.org/builders/auto-mac-64-nopt-t/builds/4992), [auto-mac-64-opt](http://buildbot.rust-lang.org/builders/auto-mac-64-opt/builds/4963), [auto-win-32-nopt-t](http://buildbot.rust-lang.org/builders/auto-win-32-nopt-t/builds/4657), [auto-win-32-opt](http://buildbot.rust-lang.org/builders/auto-win-32-opt/builds/4642), [auto-win-64-nopt-t](http://buildbot.rust-lang.org/builders/auto-win-64-nopt-t/builds/3452), [auto-win-64-opt](http://buildbot.rust-lang.org/builders/auto-win-64-opt/builds/3476)
